File: app/src/main/java/com/ShaHar91/poems/data/models/Category.java
package com.shahar91.poems.data.models;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Category extends RealmObject {
    @PrimaryKey
    private String _id;
    private String name;

    public Category() {
    }

    public Category(String _id, String name) {
        this._id = _id;
        this.name = name;
    }

    public String getId() {
        return _id;
    }

    public void setId(String _id) {
        this._id = _id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/ShaHar91/poems/data/models/Category.kt
package com.shahar91.poems.data.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import be.appwise.room.BaseEntity
import com.shahar91.poems.data.DBConstants

@Entity(tableName = DBConstants.CATEGORY_TABLE_NAME)
data class Category(
    @PrimaryKey
    @ColumnInfo(name = DBConstants.COLUMN_ID_CATEGORY)
    override var id: String = "",
    var name: String = ""
) : BaseEntity